internal/storage/v1/api/spanstore/interface.go

```diff
@@ -18,6 +18,7 @@ var ErrTraceNotFound = errors.New("trace not found")
 // Writer writes spans to storage.
 type Writer interface {
 	WriteSpan(ctx context.Context, span *model.Span) error
+	WriteDependencies(ctx context.Context, ts time.Time, dependencies []model.DependencyLink) error
```

Comment:
this should be in a separate interface, spanstore.Writer is not required to support dependencies.

also it should be in the v2 storage, no reason to update v1.